src/engine/command.cpp

```diff
@@ -532,6 +532,72 @@ char *svariable(const char *name, const char *cur, char **storage, identfun fun,
     return newstring(cur);
 }
 
+// Flags that can be used by CubeScript when defining variables.
+static constexpr ushort EXPOSED_FLAGS = IDF_HEX | IDF_PERSIST | IDF_READONLY | IDF_TEXTURE;
+
+// Expose acceptable flags as variables.
+VAR(0, idfbithex, 1, IDF_HEX, -1);
+VAR(0, idfbitpersist, 1, IDF_PERSIST, -1);
+VAR(0, idfbitreadonly, 1, IDF_READONLY, -1);
+VAR(0, idfbittexture, 1, IDF_TEXTURE, -1);
+
+// Define an integer variable.
+ICOMMAND(0, defvar, "siiiis", (const char *name, int *min, int *cur, int *max, int *flags, const char *code_on_changed), {
+    if(idents.access(name))
+    {
+        debugcode("\frCannot redefine %s as a variable", name);
+    }
+    else
+    {
+        // Use only flags exposed to CubeScript.
+        ushort filtered_flags = ((*flags) & EXPOSED_FLAGS);
+
+        // Allocate storage for the variable.
+        int *storage = new int(*cur);
```

Comment:
This cries "potential memory leak" all over the place. I'll have to look into `ident` resp. `addident` to see whether this is ever cleaned up.

Reply:
These never get deleted; C++ variables use global variables as storage.